src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9839,7 +9732,18 @@ internal BracketedArgumentListSyntax ParseBracketedArgumentList()
             SyntaxKind closeKind)
         {
             bool isIndexer = openKind == SyntaxKind.OpenBracketToken;
-            var open = this.EatToken(openKind);
+
+            if (this.CurrentToken.Kind == SyntaxKind.OpenParenToken ||
```

Comment:
>if (this.CurrentToken.Kind == SyntaxKind.OpenParenToken || [](start = 12, length = 58)

Consider adding an assert that
- ```openKind``` is either SyntaxKind.OpenParenToken or SyntaxKind.OpenBracketToken
- ```closeKind``` is either SyntaxKind.CloseParenToken or SyntaxKind.CloseBracketToken